Container mode (#197)
dcermak/rpm-spec-language-server: README.rst

``~/.cache/rpm/spec.md``.

Container Mode
==============

> The rpm-spec-language-server supports an **experimental** container mode. In
> this mode, it is launched inside a container with your package directory mounted
> into the running container. This allows you to have access to rpm macros from a
> distribution different than your current one.

Reviewer: ```suggestion
The rpm-spec-language-server is a server that supports an **experimental** container mode. In
this mode, the server is launched inside a container with a package directory mounted
into the running container. This allows you to have access to rpm
distribution different than your current one.
```

> The container mode currently supports having only one package open and rpm spec
> file **must** be in the toplevel directory. Additionally, the server **must**
> communicate via tcp. Due to that, you might have to reconfigure your lsp-client,
> if it assumes to communicate via stdio.

Reviewer: ```suggestion
The container mode can currently handle only one package open. The rpm spec
file **must** be in the top-level directory. Additionally, the server **must**
communicate via TCP. This means that you might have to reconfigure your lsp-client,
if it assumes to communicate via stdio.
```

> To invoke the container mode with podman, proceed as follows:

Reviewer: ```suggestion
To enable the container mode with Podman, proceed as follows:
```

.. code-block:: shell-session

   $ cd ~/path/to/my/package
   $ # ensure that the spec file is in the current working directory!
   $ podman container runlabel run \
         ghcr.io/dcermak/rpm-spec-lang-server:$distri

where you replace ``$distri`` with one of ``tumbleweed``, ``fedora`` or
``centos``.

> If you want to use docker, then you have to obtain the exact launch command via:
> 

Reviewer: ```suggestion
To use Docker, get the exact launch command as shown below:

```

.. code-block:: shell-session

   $ docker inspect -f '{{index .Config.Labels "run"}}' \
         ghcr.io/dcermak/rpm-spec-lang-server:$distri | \
         sed -e 's/podman/docker/' -e 's|$IMAGE|ghcr.io/dcermak/rpm-spec-lang-server:$distri|'

> where you have to once again replace ``$distri`` with the distribution that you
> wish to use.

Reviewer: ```suggestion
In the example above, replace ``$distri`` with the desired distribution.
```
